aikau/src/main/resources/alfresco/documentlibrary/_AlfHashMixin.js

```diff
@@ -148,25 +148,41 @@ define(["dojo/_base/declare",
        * @return {boolean}
        */
       payloadContainsUpdateableVar: function alfresco_documentlibrary__AlfHashMixin__payloadContainsUpdateableVar(payload, updateInstanceValues) {
-
          var containsUpdateableVar = false;
 
          // No hashVarsForUpdate - return true
          if(!this.hashVarsForUpdate || this.hashVarsForUpdate.length === 0)
          {
-            return true;
+            containsUpdateableVar = true;
          }
-         
-         // Iterate over the keys defined in hashVarsForUpdate - return true if the payload contains one of them
-         for(var i=0; i < this.hashVarsForUpdate.length; i++)
+         else
          {
-            if(this.hashVarsForUpdate[i] in payload)
+            // Iterate over the keys defined in hashVarsForUpdate - return true if the payload contains one of them
+            for(var i=0; i < this.hashVarsForUpdate.length; i++)
             {
-               if (updateInstanceValues === true)
+               if(this.hashVarsForUpdate[i] in payload)
                {
-                  this[this.hashVarsForUpdate[i]] = payload[this.hashVarsForUpdate[i]];
+                  if (updateInstanceValues === true)
+                  {
+                     this[this.hashVarsForUpdate[i]] = payload[this.hashVarsForUpdate[i]];
```

Comment:
Do we want to delete values if we get undefined/null coming through, or are we happy just persisting exactly what we receive?